### PR TITLE
Add metric to track number ssh connect attempts

### PR DESCRIPTION
**Pull request by rcanderson23**
This adds the metric `teleport_connect_to_node_attempts_total`
```
# HELP teleport_connect_to_node_attempts_total Number of ssh connect attempts
# TYPE teleport_connect_to_node_attempts_total counter
teleport_connect_to_node_attempts_total 4
```

This is added so that we can track percentage of successful connections as we already track failed attempts with `failed_connect_to_node_attempts_total`

### CODE REVIEW

**Review by rosstimothy — APPROVED** [review]

File: lib/srv/regular/proxy.go [hunk 1]
```diff
@@ -59,7 +59,15 @@ var ( // failedConnectingToNode counts failed attempts to connect to a node
 		},
 	)
 
-	prometheusCollectors = []prometheus.Collector{proxiedSessions, failedConnectingToNode}
+	connectingToNode = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      teleport.MetricConnectToNodeAttempts,
+			Help:      "Number of ssh connection attempts to a node.",
```
Comment (zmb3) [inline lib/srv/regular/proxy.go:66]:
```suggestion
			Help:      "Number of SSH connection attempts to a node.",
```

**Review by zmb3 — APPROVED** [review]
I'm accustomed to seeing related metrics like this have names that are more similar.

I recognize we're limited by the original metric, but ideally we would have something like:

```
teleport_connect_to_node_attempts_failed
teleport_connect_to_node_attempts_total
```

Since we don't want to change the original metric, I wonder if we could at least update the `Help` field or the docs so that these metrics mention each other.

**Review by xacrimon — APPROVED** [review]

File: lib/srv/regular/proxy.go [hunk 2]
```diff
@@ -55,11 +55,19 @@ var ( // failedConnectingToNode counts failed attempts to connect to a node
 	failedConnectingToNode = prometheus.NewCounter(
 		prometheus.CounterOpts{
 			Name: teleport.MetricFailedConnectToNodeAttempts,
-			Help: "Number of failed attempts to connect to a node",
+			Help: "Number of failed SSH connection attempts to a node. Used with `teleport_connect_to_node_attempts_total` to get failure rate.",
```
Comment (ptgott) [inline lib/srv/regular/proxy.go:58]:
```suggestion
			Help: "Number of failed SSH connection attempts to a node. Use with `teleport_connect_to_node_attempts_total` to get the failure rate.",
```

File: lib/srv/regular/proxy.go [hunk 3]
```diff
@@ -55,11 +55,19 @@ var ( // failedConnectingToNode counts failed attempts to connect to a node
...
 		},
 	)
 
-	prometheusCollectors = []prometheus.Collector{proxiedSessions, failedConnectingToNode}
+	connectingToNode = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: teleport.MetricNamespace,
+			Name:      teleport.MetricConnectToNodeAttempts,
+			Help:      "Number of SSH connection attempts to a node. Used with `failed_connect_to_node_attempts_total` to get failure rate.",
```
Comment (ptgott) [inline lib/srv/regular/proxy.go:66]:
```suggestion
			Help:      "Number of SSH connection attempts to a node. Use with `failed_connect_to_node_attempts_total` to get the failure rate.",
```

**Review by ptgott — APPROVED** [review]
Left a couple of very minor nits.

**Pull request merged**